src/views/catalog/customize/components/CustomizeForms/useCustomizeFormSubmit.ts

```diff
@@ -96,9 +96,9 @@ export const useCustomizeFormSubmit = ({
         vmDraft.metadata.namespace = ns || DEFAULT_NAMESPACE;
         vmDraft.metadata.labels[LABEL_USED_TEMPLATE_NAME] = processedTemplate.metadata.name;
         vmDraft.metadata.labels[LABEL_USED_TEMPLATE_NAMESPACE] = template.metadata.namespace;
-
-        vmDraft.spec.template.spec.domain.cpu.cores = vm.spec.template.spec.domain.cpu.cores;
-        vmDraft.spec.template.spec.domain.memory.guest = vm.spec.template.spec.domain.memory.guest;
+        const { cpu, memory } = getMemoryCPU(vm);
+        vmDraft.spec.template.spec.domain.cpu.cores = cpu?.cores;
```

Comment:
in some places in the codebase like exactly here, we don't wanna "?" to add. We want to let the UI get broken if the resource is not there. Because it should ALWAYS be there, in this place of the code, and if it is not there, something in the code around is wrong. We don't wanna not to see the the error we should see Due to "?" here, the UI may act like everything is working in case if not, and that is bad. Worse than showing an error.